Add ThirdPartyLicenses.csv with separate tools

felixfbecker opened the pull request:
Different approach to https://github.com/sourcegraph/sourcegraph/pull/9554 with separate tools for JS and Go. Includes the deps for JS.

@tsenart you can push to this branch to add a command that outputs CSV for Go.

I left out the CI up-to-date check for now because it would break Renovate PRs completely, but it _will_ check in CI for any GPL licenses being present (using the latest yarn.lock).

@sourcegraph/code-intel I made sure devDependencies don't actually end up in the Docker images since we don't check their licenses, some help in a quick smoke test would be appreciated 🙂

felixfbecker commented on dev/licenses.sh:
```diff
@@ -0,0 +1,19 @@
+
+set -euf -o pipefail
+
+FAIL_ON='GPL-1.0-only;GPL-1.0-or-later;GPL-2.0-only;GPL-2.0-or-later;GPL-3.0-only;GPL-3.0-or-later'
```
@dadlerj please let me know if I should add any other licenses here from  https://spdx.org/licenses/

tsenart commented:
> @tsenart you can push to this branch to add a command that outputs CSV for Go.

I don't currently have a ready to use tool that can output this easily without adding a whole set of extra dependencies to our developer environment (e.g. Ruby). So that @dadlerj has this license list available I think we should likely hold off from fully integrating these checks and generation into CI, and instead, just generate the list manually.

tsenart commented:
@dadlerj: Question: Do we need to report licenses of transitive dependencies, and, also, licenses of infrastructure components that are shipped with Sourcgraph, like prometheus, grafana, zoekt, etc?

dadlerj commented:
@tsenart theoretically I believe the answer to both of those is "yes", as they're all bundled into our product.

Prometheus/Grafana in particular may have special handling given they're standalone products that we just ship alongside our own though...

I'll check with the lawyers to confirm.

tsenart commented:
@felixfbecker: I added a `GoLicenses.csv` file. It's not feasible to run the tool I used to generate that file in CI because it doesn't work with every dependency and it is also slow, so I think we should just manually generate these for now.

felixfbecker commented:
Thanks @tsenart! I renamed the other file accordingly. It may still be useful to have a shell script or docs on how to generate the Go file, but I think Core Services should own this file, so whatever works for you :) I added CODEOWNER entries for the two files.

Pending approval from @sourcegraph/code-intel on the Dockerfile changes and from @dadlerj on the license block list I think this is ready to merge ✨
